### PR TITLE
Move the KMS integration imports into the binary entrypoints

### PR DESCRIPTION
**Pull request by mattmoor**
#### Ticket Link
Related: https://github.com/sigstore/sigstore/issues/384
Related: https://github.com/sigstore/sigstore/issues/386

#### Release Note

```release-note
BREAKING: if you use cosign as an SDK and rely on KMS integrations, you will need to start importing the integrations you would like to support.
```

### CODE REVIEW

**Comment by mattmoor** [comment]
cc @imjasonh

**Review by imjasonh — APPROVED** [review]
🙌

File: cmd/cosign/policy_webhook/main.go [hunk 1]
```diff
@@ -34,6 +34,14 @@ import (
...
 	"github.com/sigstore/cosign/pkg/apis/cosigned/v1alpha1"
 	"github.com/sigstore/cosign/pkg/reconciler/clusterimagepolicy"
+
+	// Register the provider-specific plugins
+	_ "github.com/sigstore/sigstore/pkg/signature/kms/aws"
+	_ "github.com/sigstore/sigstore/pkg/signature/kms/azure"
+	_ "github.com/sigstore/sigstore/pkg/signature/kms/gcp"
+	_ "github.com/sigstore/sigstore/pkg/signature/kms/hashivault"
+	// TODO(mattmoor): Not sure why this is in the final binary...
+	// _ "github.com/sigstore/sigstore/pkg/signature/kms/fake"
```
Comment (haydentherapper) [inline cmd/cosign/policy_webhook/main.go:44]: `fake` was added for testing, it should just be required in a few test files

**Review by dlorenc — APPROVED** [review]

File: cmd/cosign/policy_webhook/main.go (hunk 1 above)
Comment (mattmoor) [reply cmd/cosign/policy_webhook/main.go:44]: Yeah, I suspected that, but it was linked from `pkg/reconciler/...` (not in test code!), so I wanted to run this through CI to figure out what tests are expecting this.  I'm going to guess now that this was bad copy/pasta and just remove this instance.

Comment (mattmoor) [reply cmd/cosign/policy_webhook/main.go]: This is removed in the commit I just pushed.

**Pull request merged**